crates/sui-graphql-rpc/src/types/checkpoint.rs

```diff
@@ -199,6 +199,21 @@ impl Checkpoint {
         Ok(stored.map(|stored| Checkpoint { stored }))
     }
 
+    /// Look up a `Checkpoint` in the database, filtered by either sequence number or digest. If
+    /// both filters are supplied they will both be applied. If none are supplied, the latest
+    /// checkpoint is fetched.
+    pub(crate) async fn query_latest(db: &Db) -> Result<Self, Error> {
```

Comment:
curious, what happens if one filter is ok but the other is wrong (e.g., an invalid checkpoint)? Will it error or will it use the correct one?

Reply:
ah oops this doc comment is not applicable. but it will apply both filters, resolving in a null result. Since part of the filter is invalid, we should not return a meaningful response